presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorSplitManager.java

```diff
@@ -17,17 +17,49 @@
 import com.facebook.presto.spi.ConnectorSplitSource;
 import com.facebook.presto.spi.ConnectorTableLayoutHandle;
 
+import static java.util.Objects.requireNonNull;
+
 public interface ConnectorSplitManager
 {
     ConnectorSplitSource getSplits(
             ConnectorTransactionHandle transactionHandle,
             ConnectorSession session,
             ConnectorTableLayoutHandle layout,
-            SplitSchedulingStrategy splitSchedulingStrategy);
+            SplitSchedulingContext splitSchedulingContext);
 
     enum SplitSchedulingStrategy
     {
         UNGROUPED_SCHEDULING,
         GROUPED_SCHEDULING,
     }
+
+    public static class SplitSchedulingContext
+    {
+        private final SplitSchedulingStrategy splitSchedulingStrategy;
+        private final boolean schedulerUsesHostAddresses;
+
+        /**
+         * @param splitSchedulingStrategy the method by which splits are scheduled
+         * @param schedulerUsesHostAddresses whether host addresses are take into account
+         * when choosing where to schedule remotely accessible splits. If this is false,
+         * the connector can return an empty list of addresses for remotely accessible
+         * splits without any performance loss.  Non-remotely accessible splits always
+         * need to provide host addresses.
+         */
+        public SplitSchedulingContext(SplitSchedulingStrategy splitSchedulingStrategy, boolean schedulerUsesHostAddresses)
```

Comment:
Thanks for the explanation in comments! It looks great.

Speaking of the variable name, I am not sure how easy it is for a connector developer to understand it (without looking into the comment). My first impression is "what's scheduler?" . 

I am also thinking something like `hostAddressesWillBeUsed`, but looks really bad as variable name 😞 

Thoughts? @highker

Reply:
`hostAddressesAreUseful` is okay too. I guess it's better because it's more focused on what the connector writers need to know?  I don't think either will be super clear without reading the docs.

Reply:
`schedulerUsesHostAddresses` sounds good to me :D